src/nidmm/system_tests/test_system_nidmm.py

```diff
@@ -7,16 +7,20 @@
 
 
 @pytest.fixture(scope='function')
-def device_info(request):
+def device_info(request, name):
     device_info = {}
-    device_name = "unknown"
+    device_name = name
     device_sn = "unknown"
     try:
         with nimodinst.Session('nidmm') as session:
             if len(session) > 0:
-                device_name = session.device_name[0]
-                device_sn = session.serial_number[0]
-
+                for x in range(0, len(session)):
```

Comment:
This is where having some static documentation would help. Right now the only way to know this is by looking at the example.

You can write this loop as `for device in session:`

Then you just use `device_name = device.device_name`